### PR TITLE
Fix edge case for 'add missing labels' gha

### LINKED ISSUE #2227: GitHub Actions: Fix edge case scenario on 'add missing labels' GHA
### Overview
As an organization, we want to change our "size: good first issue" label to "good first issue" so that it can be referenced by GitHub and other organizations. However, doing so will break the GitHub Action:  "Add Missing Labels to Issues." Update and fix the edge case scenario for the GitHub Action so that changing the label won't break the action.

### Action Items
- [x] Understand how the workflow for the GitHub Action works
    - [x] GithubAction workflow file: [add-missing-labels-to-issues.yml](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/add-missing-labels-to-issues.yml)
    - [x] Accompanying JS: [check-labels.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/issue-labels/check-labels.js)
- [x] Take a look at the [checkLabels function](https://github.com/hackforla/website/blob/447b209b9ceacd17f850b291373828be2012a74a/github-actions/issue-labels/check-labels.js#L58)
    - [x] Update the conditional so it takes into account a 'good first issue' label when dealing with the required label 'Size'
    - [x] Comment the code so other developers can understand what the change does
    - **Note:** A fix to the code will be added below under "One Possible Fix" if a hint is needed to solve the problem. However, it is encouraged to understand what the problem is and come up with the solution without checking this hint.
- [x] Test that the change in the code works in your own fork of the repository

<details>
<summary>One Possible Fix</summary>

```
function checkLabels(labels) {
  let labelsToAdd = []

  REQUIRED_LABELS.forEach((requiredLabel, i) => {
    const regExp = new RegExp(`\\b${requiredLabel}\\b`, 'gi')
    const isLabelPresent = labels.some(label => {
      if (label === 'good first issue' && requiredLabel === 'Size') {
        return true
      }

      return regExp.test(label)
    })

    if (isLabelPresent === false){
      labelsToAdd.push(LABEL_MISSING[i])
    }
  })
  
  return labelsToAdd
}
```

</details>

### Resources/Instructions
Never done GitHub actions? [Start here!](https://docs.github.com/en/actions)

#### Additional resources:
[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)
[Add Missing Labels To Issues](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/add-missing-labels-to-issues.yml)
[check-labels.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/issue-labels/check-labels.js)

### PR DESCRIPTION
**Pull request by macho-catt**
Fixes #2227

### What changes did you make and why did you make them ?

  - Fix edge case so that when the label "good first issue" is added, the action will consider it as a "size" label
  - Github action now acts as follows:
      - If there are no labels, three missing labels will be added (size missing, feature missing, role missing)
      - If good first issue is the only one added, two missing labels will be added (feature missing, role missing)
      - If good first issue is not added and a size label is not provided, size missing label will be added
      - If all three labels are added, a comment from the bot will be posted

### Screenshots of Issue Creation on my branch

<details>
<summary>No labels</summary>

no-labels

</details>

<details>
<summary>Only size label provided: good first issue</summary>
  
1-label

</details>

<details>
<summary>Size label not provided</summary>
  
no-size

</details>

<details>
<summary>All labels provided</summary>
  
all-labels

</details>

This will allow the current label, "Size: good first issue" to be turned into "good first issue" as per #2301

### CODE REVIEW

**Comment by macho-catt** [comment]
Also removed unneeded json file from a different branch.

File: github-actions/issue-labels/check-labels.js [hunk 1]
```diff
@@ -60,7 +60,14 @@ function checkLabels(labels) {
 
   REQUIRED_LABELS.forEach((requiredLabel, i) => {
     const regExp = new RegExp(`\\b${requiredLabel}\\b`, 'gi')
-    const isLabelPresent = labels.some(label => regExp.test(label))
+    const isLabelPresent = labels.some(label => {
+      // The label 'good first issue' fulfills the size requirement for labels
+      if (label === 'good first issue' && requiredLabel === 'Size') {
+        return true
```
Comment (Aveline-art) [inline github-actions/issue-labels/check-labels.js]: Consider whether you want to include 'good first issue' as part of an array on top of the file, perhaps called 'sizeExceptions'.

**Review by Aveline-art — DISMISSED** [review]
Feel free to merge this after considering my comment below. I can attest that this works as intended.

File: github-actions/issue-labels/check-labels.js (hunk 1 above)
Comment (macho-catt) [reply github-actions/issue-labels/check-labels.js]: good idea, I can do that.

**Review by Aveline-art — APPROVED** [review]
Looks good!

**Pull request merged**